libkbfs/mdserver_disk.go

```diff
@@ -383,6 +383,50 @@ func (md *MDServerDisk) GetForTLF(ctx context.Context, id tlf.ID,
 	return tlfStorage.getForTLF(ctx, session.UID, bid)
 }
 
+// GetForTLFByTime implements the MDServer interface for MDServerDisk.
+func (md *MDServerDisk) GetForTLFByTime(
+	ctx context.Context, id tlf.ID, serverTime time.Time) (
+	*RootMetadataSigned, error) {
+	if err := checkContext(ctx); err != nil {
+		return nil, err
+	}
+
+	session, err := md.config.currentSessionGetter().GetCurrentSession(ctx)
+	if err != nil {
+		return nil, kbfsmd.ServerError{Err: err}
+	}
+
+	tlfStorage, err := md.getStorage(id)
+	if err != nil {
+		return nil, err
+	}
+
+	rmd, err := tlfStorage.getForTLF(ctx, session.UID, kbfsmd.NullBranchID)
+	if err != nil {
+		return nil, err
+	}
+
+	for rmd.untrustedServerTimestamp.After(serverTime) {
+		nextRev := rmd.MD.RevisionNumber() - 1
```

Comment:
Please comment that for the disk MDServer we're doing a linear search.

Reply:
Sure -- where do you think the comment should be?  Right here?

Reply:
I'd say just above the loop.